qiskit/visualization/pulse/qcstyle.py

```diff
@@ -128,12 +130,12 @@ def __init__(self, figsize: Tuple[float, float] = (10, 12),
 
 class PulseStyle:
     """Style sheet for Qiskit-Pulse sample pulse drawer."""
-    def __init__(self, figsize: Tuple[float, float] = (7, 5),
-                 title_font_size: int = 18,
-                 wave_color: List[str] = None,
-                 bg_color: str = None,
-                 num_points: int = None,
-                 dpi: int = None):
+    def __init__(self, figsize: Tuple[float, float] = mpl.rcParams['figure.figsize'],
+                 title_font_size: Union[int, str] = mpl.rcParams['figure.titlesize'],
+                 wave_color: List[str] = mpl.rcParams['axes.prop_cycle'].by_key()['color'][:2],
```

Comment:
What functionality do we want here? I can imagine one of three options for colors,

1. Hardcode colors as before.
2. By default take the first two colors in the prop cycle list.
3. Query the prop cycler state and take the next two colors.

What you see implemented above is (2). I can do (3) if we want, but (2) may look nicer and more homogenous amongst other plots.

Reply:
I think my hope was to keep our default styling but make parameters such as figure size configurable via rcParams. Would this be possible?

Reply:
Yea i think it's possible.  Sorry, i misinterpreted the original ask.  Will post a PR with the fix.  Basically if we did dthat, then there would be no need for these objects at all right?

Reply:
I'll double check but just in case there are any paramters that are both inherited from the same rcparam, i'll have to break them out into a config or something.

Reply:
If the default is set to what we had before and the user has the ability to put `None`, then that (in most cases) should just revert back to the rcparams.

Reply:
I worried about [setting mutable as default argument](https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/). If you need to set color array as default we may need to deepcopy the argument.

Reply:
See the updated version of this PR.  I already replaced all the mutables with immulatbles.

Reply:
You are correct in that the mutable versions are bad, and it's caught by the linter anway.

Reply:
https://github.com/Qiskit/qiskit-terra/pull/4470/files#diff-77c5f33032e5653206d40605a8283f6fR27